src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/JsonSerializerContextTests.cs

```diff
@@ -310,6 +310,187 @@ public static IEnumerable<object[]> GetCombiningContextsData()
             static object[] WrapArgs<T>(T value, string expectedJson) => new object[] { value, expectedJson };
         }
 
+        [Fact]
+        [ActiveIssue("Should src-gen'd JsonSerializerContext.GetTypeInfo(Options) implementations be sealed?")]
+        public static void SourceGeneratedContextsCanBeModified_JsonSerializerOptionsOverload()
+        {
+            Test(new EditedSourceGenContext());
+            Test(new EditedSourceGenContext_Metadata());
+            Test(new EditedSourceGenContext_Serialization(), isFastPath: true);
+
+            static void Test(IJsonTypeInfoResolver resolver, bool isFastPath = false)
+            {
+                // This options instance here differs from the instance the context/resolver
+                // was initialized with. This means that user overrides are not consulted to offer
+                // a type-info instances. This is different from DefaultJsonTypeInfo behavior where
+                // overrides of `GetTypeInfo(Type type, JsonSerializerOptions options)` are respected.
+                // Should we seal `JsonSerializerContext.GetTypeInfo(Options)` since it might be tricky
+                // to respect it?
+                JsonSerializerOptions options = new() { TypeInfoResolver = resolver };
```

Comment:
This options instance is different from the instance the context/resolver was initialized with. This means that user `JsonSerializerContext.GetTypeInfo(Options)` overrides are not consulted to offer type-info instances. This is different from DefaultJsonTypeInfo behavior where overrides of `GetTypeInfo(Type type, JsonSerializerOptions options)` are respected.

Should we seal `JsonSerializerContext.GetTypeInfo(Options)` since it might be tricky to respect it? (I believe there might also be options due to the new `JsonSerializerOptions` pooling/reuse mechanism)

Reply:
I'm pretty certain we have test coverage for that somewhere.

Reply:
The issue is that we always go into the `else` block when `JsonSerializerOptions`-based overloads are used.

Reply:
From a performance standpoint? These would be cached by the `JsonSerializerOptions` instance regardless.

Reply:
If the user manually overrides `GetTypeInfo` with `JsonSerializable` annotations added, then it will conflict with the source generated implementation and fail to compile. If it's manually overridden with source generation disabled, then there is no interface implementation for `IJsonTypeInfoResolver` and it will use the base class semantics.

Reply:
i.e. 

https://github.com/dotnet/runtime/blob/bce23abb964d81b6d626f2a2b435549636769a1d/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs#L122-L130